botkit.d.ts

```diff
@@ -0,0 +1,427 @@
+        startConversation(message: Message, cb: (err: Error, convo: Conversation) => void): never;
+        createConversation(message: Message, cb: (err: Error, convo: Conversation) => void): never;
+        private _startDM(task: Task, user_id: string, cb: Function): any;
+        send(message: Message, cb: (err?: string | Error) => void): never;
+        replyAcknowledge(cb: (err?: string | Error) => void): never;
+        replyPublic(src: any, resp: string, cb: (err?: string | Error) => void): never;
+        replyPublicDelayed(src: any, resp: string, cb: (err?: string | Error) => void): never;
+        replyPrivate(src: any, resp: string, cb: (err?: string | Error) => void): never;
+        replyPrivateDelayed(src: any, resp: string, cb: (err?: string | Error) => void): never;
+        replyInteractive(src: any, resp: string, cb: (err?: string | Error) => void): never;
+        reply(src: Message, resp: string);
+        reply(src: Message, resp: string, cb?: (err?: string | Error) => void): never;
+        say(message: Message, cb?: (err?: string | Error) => void): never;
+        startTyping(src: any): never;
+        replyWithTyping(src: any, resp: string, cb: (err?: string | Error) => void): never;
+        replyAndUpdate(src: any, resp: string, cb: (err?: string | Error) => void): never;
+        findConversation(message: Message, cb?: () => void): never;
+    }
+    type Status = 'new' | 'completed' | 'active' | 'inactive' | 'ending' | 'stopped';
+    class Conversation {
+        messages: Message[];
+        sent: Message[];
+        transcript: any[];
+        context: {
+            user: User;
+            channel: Channel;
+            bot: Bot;
+        };
+        events: { [event: string]: Function }
+        vars: any;
+        status: Status;
+        task: Task;
+        source_message: Message;
+        handler: any;
+        responses: any;
+        capture_options: any;
+        startTime: Date;
+        lastActive: Date;
+
+        constructor(task: Task, message: Message);
+        collectResponse(key, value): never;
+        capture(response): never;
+        handle(message: Message): void;
+        setVar(field, value): never;
+        activate(): never;
+        /**
+         * active includes both ACTIVE and ENDING
+         * in order to allow the timeout end scripts to play out
+         **/
+        isActive(): boolean;
+        deactivate(): never;
+        say(message: string | Message): never;
+        sayFirst(message: string | Message): never;
+        on(event: string, cb: Function): never;
+        trigger(event: string, data): void;
+        /**
+         * proceed to the next message after waiting for an answer
+         */
+        next(): never;
+        repeat(): never;
+        silentRepeat(): never;
+        addQuestion(message: Message, cb: Function, capture_options, thread): never;
+        ask(message: Message, cb: Function, capture_options): never;
```

Comment:
Using this type definition, found some odd types in Conversation class:
- `cb` might be either `Function` or `array of mapping of keyword to callback`.
- `message` might be either `string` or `Message`.
See: https://github.com/howdyai/botkit#using-conversationask-with-an-array-of-callbacks
Example: https://github.com/howdyai/botkit#using-conversationask-with-an-array-of-callbacks

It works for my project:
```diff
diff --git a/botkit.d.ts b/botkit.d.ts
index eac474b..ffd615a 100644
--- a/botkit.d.ts
+++ b/botkit.d.ts
@@ -373,8 +373,8 @@ declare module botkit {
         next(): never;
         repeat(): never;
         silentRepeat(): never;
-        addQuestion(message: Message, cb: Function, capture_options, thread): never;
-        ask(message: Message, cb: Function, capture_options): never;
+        addQuestion(message: string | Message, cb: Function | {pattern?: string | RegExp, callback: Function, default?: boolean}[], capture_options, thread): never;
+        ask(message: string | Message, cb: Function | {pattern?: string | RegExp, callback: Function, default?: boolean}[], capture_options): never;
         addMessage(message: Message, thread): never;
         /**
          * how long should the bot wait while a user answers?
```